### PR TITLE
Add link to connect route with user

### PR DESCRIPTION
**Pull request by nd0761**
Add class for Link
Change existing files accordingly

### CODE REVIEW

**Comment by nd0761** [comment]
i accidentally pulled master into that branch, so only first and two last commits make any sense

**Review by ioannanedelcu — APPROVED** [review]

File: src/main/webapp/mapScript.js [hunk 1]
```diff
@@ -181,13 +184,25 @@ async function createRoute() {
...
     alert(
       "Route successfully created!\nYou can see new created routes on your profile page!"
     );
   }
   loadRoutes();
 }
 
+function updateShareList() {
+  // For now lets assume that Friend Code is ID in datastore.
+  // For now letc assume that all co-owners are editors.
```
Comment (conradgrobler) [inline src/main/webapp/mapScript.js:197]: s/letc/lets/

File: src/main/java/com/google/sps/servlets/RoutesStoring.java [hunk 2]
```diff
@@ -79,14 +80,27 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
...
       datastore.put(routeEntity);
+      for (long userId : editorsArray) {
+        Entity linkEntity = new Entity("Link", KeyFactory.createKey("User", userId));
+        linkEntity.setProperty("routeId", routeEntity.getKey().getId());
+        linkEntity.setProperty("type", 2);
+        // Add entity for editor.
+        datastore.put(linkEntity);
+      }
+
+      Entity linkEntity = new Entity("Link", userEntity.getKey());
```
Comment (conradgrobler) [inline src/main/java/com/google/sps/servlets/RoutesStoring.java:99]: nit: "Link" is not very descriptive, does not match the associated java class name, and might be confusing if you add addtional links between other entities in future.

File: src/main/java/com/google/sps/data/RouteUserLink.java [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+  private Long userId;
+  private Long routeId;
+
+  // 1 - owner, 2 - editor, 3 - viewer
+  private int type;
+
+  public RouteUserLink(Long newUserId, Long newRouteId, int type) {
+    userId = newUserId;
+    routeId = newRouteId;
+
```
Comment (conradgrobler) [inline src/main/java/com/google/sps/data/RouteUserLink.java:28]: nit: I think the extra blank line does not improve readability and would be simpler without it. Same applie on L35.

**Review by conradgrobler — APPROVED** [review]

**Pull request merged**